### DISCUSSION TITLE
Need some help caching tiles.

### DISCUSSION THREAD

**Discussion by mattiacerutti** [JaffaKetchup/flutter_map_tile_caching#85] [Q&A] [answered]
Hello, im building a very basic application with a map and my live location, im currently using mapbox as a static tile provider api. I noticed the map was reloading everytime i zoomed in or out causing my app to have a lot of gray squares. Searched up a little in the internet and found this really intresting plugin, read all the documentation but didn't really understand how to implement it in my app. I tried looking in the example on this repo but it's too complex for my case and there are no online guides. Can someone please help me figure this out? Would really help!

Thanks

**Comment by JaffaKetchup** [comment, marked as answer]
Hey @Clucch, I'm happy to help, but you'll need to make a start yourself.

The example app isn't intended to be followed for new projects/implementations, it serves more as a reference on how you might implement surrounding/supporting features using the FMTC API.

Start with https://fmtc.jaffaketchup.dev/get-started/basic-terminology, then head over to https://fmtc.jaffaketchup.dev/usage/initialisation.
For basic usage, you can ignore most of Using Roots and Stores, although it's recommended to implement at least some simple Management for your users. You'll need to use https://fmtc.jaffaketchup.dev/usage/integration to connect FMTC to your map instance.

Please keep trying and let me know if you get stuck, so I can help with a more specific issue.

**Reply by mattiacerutti** [reply]
So, i've integrated the FMTC to the flutter map and it seems to be working pretty well. The only problem is that if im browsing my map and then i suddently disable my internet connection, it goes well for the few tiles cached, but as soon as i move i get an exception (because internet is missing of course). How can i catch this error in order to maybe display something or just show unloaded tiles instead of crashing the app?

I also read in the docs that there is this setting called `maxStoreLength` and as mentioned it could get pretty heavy on performance if set too high, but i can't seem to get a scale of measurment of how much is "too high". 

Thank you for your kindness.

**Reply by JaffaKetchup** [reply]
@Clucch Great!

Can you post that error message here please? There's a good chance you can safely ignore it (it simply represents a fall through of all possible cases, so must throw an error, but in this case, that's perfectly safe and intended. v7 will come with a mechanism to programtically catch & silence the error, but for now, just pretend it's not there: assuming it's the one I think it is.

`maxStoreLength` is tough one to decide and really depends on the device. The issue is that it has to sort the entire tile store in order of modification time to find the oldest one to delete. A reasonable place to put it is below 5000 tiles. In v7, this issue is resolved and can be set high as you like :).

You can find more information about v7 below. Currently I would say it is almost ready for production usage, and ready for hobbyists. Any feedback would be appriciated if you can get it working (there's a prerelease on pub.dev), but no public documentation for now.
- https://fmtc.jaffaketchup.dev/roadmap
- https://github.com/JaffaKetchup/flutter_map_tile_caching/pull/78

**Reply by mattiacerutti** [reply]
Here's the error: `FMTCBrowsingError: Failed to load the tile from the cache or the network because it was missing from the cache and a connection to the server could not be established.`. Thank you for your patience, i will take a look at v7!

**Reply by JaffaKetchup** [reply]
Indeed, that error is the expected one I was talking about. No problem, and please don't hesitate to leave any feedback - it's always appriciated :)
